### PR TITLE
[APPC-3199] Implement my wallet 3 0 wallet renaming

### PR DESCRIPTION
**Pull request by G0retZ**
**What does this PR do?**

   Add wallet renaming feature.

   The wallet name gets saved in the DB only, extracted from the back up file, used as a backup file name and default to the "Main Wallet" on the wallet creation. Turned out auto migrations don't work well with CI 😔 

**Database changed?**

   Yes

**Where should the reviewer start?**

- Better by commits

**How should this be manually tested?**

  Checked that everywhere the wallet name is used. Try to rename the wallet

**What are the relevant tickets?**

  Tickets related to this pull-request: [APPC-3199](https://aptoide.atlassian.net/browse/APPC-3199)

**Code Review Checklist**

- [ ] Architecture
- [ ] Documentation on public interfaces
- [ ] Database changed?
- [ ] Database migration (if applicable)?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] Functional QA tests pass

### CODE REVIEW

File: app/src/main/res/layout/fragment_my_wallets_name.xml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      app:layout_constraintStart_toStartOf="parent"
+      app:layout_constraintTop_toBottomOf="@id/title"
+      app:textFieldColor="@color/white"
+      app:textFieldHint="@string/mywallet_choose_name_field"
+      app:textFieldType="outlined"
+      />
+  <com.asfoundation.wallet.ui.common.WalletButtonView
+      android:id="@+id/save"
+      android:layout_width="wrap_content"
+      android:layout_height="@dimen/large_button_height"
```
Comment (rodolfobcardoso) [inline app/src/main/res/layout/fragment_my_wallets_name.xml:46]: Looks like this should be small_button_height

File: app/src/main/java/com/asfoundation/wallet/my_wallets/name/NameDialogViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    fun initialState(savedStateHandle: SavedStateHandle): NameDialogState = NameDialogState(
+      savedStateHandle.get<String>(NameDialogFragment.WALLET_ADDRESS_KEY)!!,
+      Async.Success(savedStateHandle.get<String>(NameDialogFragment.WALLET_NAME_KEY)!!),
+    )
+  }
+
+  fun setName(name: String) {
+    val retainValue = NameDialogState::walletNameAsync
+    state.walletNameAsync()?.also {
+      walletInfoRepository.updateWalletName(state.walletAddress, name)
```
Comment (rodolfobcardoso) [inline app/src/main/java/com/asfoundation/wallet/my_wallets/name/NameDialogViewModel.kt:39]: Shouldn't we use a useCase/interactor here? It's ok either way, but I feel it's better if we either always use useCases or never use them at all.

**Review by rodolfobcardoso — COMMENTED** [review]
Might not be related to the PR, but the total balance in myWallets is being displayed as 3x the amount of credits. Other than that every thing is looking good

File: app/src/main/res/layout/fragment_my_wallets_name.xml (hunk 1 above)
Comment (G0retZ) [reply app/src/main/res/layout/fragment_my_wallets_name.xml:46]: Used the same as in the Terms and Conditions bottom sheet

File: app/src/main/java/com/asfoundation/wallet/my_wallets/name/NameDialogViewModel.kt (hunk 2 above)
Comment (G0retZ) [reply app/src/main/java/com/asfoundation/wallet/my_wallets/name/NameDialogViewModel.kt]: Added use case for convenience

**Comment by G0retZ** [comment]
> Might not be related to the PR, but the total balance in myWallets is being displayed as 3x the amount of credits. Other than that every thing is looking good

Fixed the bug

**Pull request merged**